plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-color-palette.tsx

```diff
@@ -78,6 +85,16 @@ export const SidebarNavigationScreenColorPalette = () => {
 	return (
 		<SidebarNavigationScreen
 			title={ title }
+			onNavigateBackClick={ () => {
+				const assemblerUrl = getNewPath(
+					{ customizing: true },
+					'/customize-store/assembler-hub',
+					{}
+				);
+
+				navigateTo( { url: assemblerUrl } );
+				navigate( SidebarNavigationAnimationDirection.Back );
+			} }
```

Comment:
This rather complex back-click handler is repeated many times, sometimes with `resetHighlightedBlockClientId()`, sometimes without. Could it be extracted into one place?

In Site Editor we ended up having the specific sidebar components not defining the back location, and the location is instead passed from above as a prop.

In the Woo case that would mean moving the handler near the `getComponentByPathParams` function, and returning components like:
```js
return <SidebarNavigationScreenHeader onNavigateBackClick={...} />
```

Reply:
Great feedback! Thanks! I addressed it with 3a80ac3c3c04dbcfd879af20b4a93da807de64b7